### PR TITLE
Add mypy to additional files

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

This PR removes two of the imports that we have set to `skip`, so we are doing some more type checking. Tagging you for review @io7m since most of the changes come from the fixtures.

## Motivation and Context

This PR has to go in first: https://github.com/ThePalaceProject/circulation/pull/867 as this one is based on it.

## How Has This Been Tested?

Running mypy

## Checklist

- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

File: api/sip/__init__.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def patron_information(self, username, password):
         except OSError as e:
             raise RemoteIntegrationException(self.server or "unknown server", str(e))
 
-    def _remote_patron_lookup(self, patron_or_patrondata):
+    def _remote_patron_lookup(self, patron_or_patrondata) -> Union[None, PatronData]:
```
Comment (io7m) [inline api/sip/__init__.py:241]: `Union[None, PatronData]` is isomorphic to `Optional[PatronData]`. I think `Optional` is clearer.

File: core/model/configuration.py [hunk 2]
```diff
@@ -329,6 +329,7 @@ class ExternalIntegration(Base):
         "Library",
         back_populates="integrations",
         secondary=lambda: externalintegrations_libraries,
+        uselist=True,
```
Comment (io7m) [inline core/model/configuration.py:332]: I don't get what happened here.

File: tests/fixtures/api_controller.py [hunk 3]
```diff
@@ -102,7 +102,7 @@ def set_base_url(self):
         base_url.value = "http://test-circulation-manager/"
 
     def circulation_manager_setup_with_session(
-        self, session: Session, overrides: ControllerFixtureSetupOverrides = None
+        self, session: Session, overrides: ControllerFixtureSetupOverrides | None = None
```
Comment (io7m) [inline tests/fixtures/api_controller.py:105]: `ControllerFixtureSetupOverrides | None` <-> `Optional[ControllerFixtureSetupOverrides]`?

File: tests/fixtures/api_controller.py [hunk 4]
```diff
@@ -168,7 +168,7 @@ def circulation_manager_setup_with_session(
         return self.manager
 
     def circulation_manager_setup(
-        self, overrides: ControllerFixtureSetupOverrides = None
+        self, overrides: ControllerFixtureSetupOverrides | None = None
```
Comment (io7m) [inline tests/fixtures/api_controller.py:171]: As above.

File: tests/fixtures/odl.py [hunk 5]
```diff
@@ -65,7 +66,7 @@ class ODLTestFixture:
     def __init__(
         self,
         db: DatabaseTransactionFixture,
-        files: ODLAPIFilesFixture,
+        files: APIFilesFixture,
```
Comment (io7m) [inline tests/fixtures/odl.py:69]: What happened here?

**Review by io7m — CHANGES_REQUESTED** [review]
Looks mostly good, just a few incredibly minor tweaks and questions.

File: tests/fixtures/api_controller.py (hunk 3 above)
Comment (jonathangreen) [reply tests/fixtures/api_controller.py:105]: Yes, those two are equivalent. This new `|` syntax is enabled by importing `from __future__ import annotations` at the top of the file.

In any file that does this future import the pyupgrade linter prefers the `|` union notion over the old `Optional[]` form, see [docs here](https://github.com/asottile/pyupgrade#pep-604-typing-rewrites).

File: tests/fixtures/odl.py (hunk 5 above)
Comment (jonathangreen) [reply tests/fixtures/odl.py:69]: Without this change, mypy gives this error: 
```
tests/fixtures/odl.py:296: error: Argument 2 to "ODL2TestFixture" has incompatible type "ODL2APIFilesFixture"; expected "ODLAPIFilesFixture"  [arg-type]
```

`ODL2TestFixture`:
https://github.com/ThePalaceProject/circulation/blob/5ecedbc042927e219a4faab3a730732756fff19c/tests/fixtures/odl.py#L285-L286

I think this is the correct change, as both `ODLAPIFilesFixture` and `ODL2APIFilesFixture` use `APIFilesFixture` as their base class.

File: api/sip/__init__.py (hunk 1 above)
Comment (jonathangreen) [reply api/sip/__init__.py]: Updated in most recent commit

**Review by io7m — APPROVED** [review]
OK, seems good!

**Pull request merged**